x/lockup/keeper/lock.go

```diff
@@ -281,6 +286,43 @@ func (k Keeper) UnlockPeriodLockByID(ctx sdk.Context, LockID uint64) (*types.Per
 	return lock, err
 }
 
+// AddTokensToLock locks more tokens into a lockup
+func (k Keeper) AddTokensToLock(ctx sdk.Context, owner sdk.AccAddress, lockID uint64, coins sdk.Coins) (*types.PeriodLock, error) {
+	lock, err := k.GetLockByID(ctx, lockID)
+	if err != nil {
+		return nil, err
+	}
+	if lock.Owner != owner.String() {
+		return nil, types.ErrNotLockOwner
+	}
+	if err := k.bk.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, coins); err != nil {
+		return nil, err
+	}
+	lock.Coins = lock.Coins.Add(coins...)
+
+	// store lock object into the store
+	store := ctx.KVStore(k.storeKey)
+	bz, err := proto.Marshal(lock)
+	if err != nil {
+		return nil, err
+	}
+	store.Set(lockStoreKey(lock.ID), bz)
+
+	// modifications to accumulation store
+	for _, coin := range lock.Coins {
+		// remove previous store for the lock ID and add again with updated value
+		k.accumulationStore(ctx, coin.Denom).Remove(accumulationKey(lock.Duration, lock.ID))
+		k.accumulationStore(ctx, coin.Denom).Set(accumulationKey(lock.Duration, lock.ID), coin.Amount)
+	}
+
+	if k.hooks == nil {
+		return lock, nil
+	}
```

Comment:
Whats the intuition for why we have this check? Is there some performance concern?

Reply:
There is an issue with app initialization script for lockup keeper. Incentives module keep lockup keeper but that keeper is not initialized with hooks because there are circular dependencies in initialization. And for temporary fix, just added the code. And lockup hooks are not used now fyi.